ocs_ci/utility/kms.py

```diff
@@ -0,0 +1,388 @@
+            )
+        cmd = f"vault secrets enable -path={self.vault_backend_path} kv"
+        out = subprocess.check_output(shlex.split(cmd))
+        if "Success" in out.decode():
+            logger.info(f"vault path {self.vault_backend_path} created")
+        else:
+            raise VaultPathCreationFailed(
+                f"Failed to create path f{self.vault_backend_path}"
+            )
+        self.vault_create_policy()
+
+    def vault_create_policy(self):
+        """
+        Create a vault policy and generate token
+
+        """
+        policy = (
+            f'path "{self.vault_backend_path}/*" {{\n'
+            f'  capabilities = ["create", "read", "update","delete"]'
+            f"\n}}\n"
+            f'path "sys/mounts" {{\n'
+            f'capabilities = ["read"]\n'
+            f"}}"
+        )
+        vault_hcl = tempfile.NamedTemporaryFile(mode="w+", prefix="test", delete=False)
+        with open(vault_hcl.name, "w") as hcl:
+            hcl.write(policy)
+
+        if not config.ENV_DATA.get("VAULT_POLICY"):
+            self.vault_policy_name = (
+                f"{constants.VAULT_DEFAULT_POLICY_PREFIX}-" f"{self.cluster_id}"
+            )
+        else:
+            self.vault_policy_name = config.ENV_DATA.get("VAULT_POLICY")
+
+        cmd = f"vault policy write {self.vault_policy_name} {vault_hcl.name}"
+        out = subprocess.check_output(shlex.split(cmd))
+        if "Success" in out.decode():
+            logger.info(f"vault policy {self.vault_policy_name} created")
+        else:
+            raise VaultPolicyCreationFailed(
+                f"Failed to create policy f{self.vault_policy_name}"
+            )
+        self.vault_path_token = self.generate_vault_token()
+
+    def generate_vault_token(self):
+        """
+        Generate a token for self.vault_policy_name
+
+        Returns:
+            str: vault token
+
+        """
+        cmd = f"vault token create -policy={self.vault_policy_name} " f"--format=json"
+        out = subprocess.check_output(shlex.split(cmd))
+        json_out = json.loads(out)
+        return json_out["auth"]["client_token"]
+
+    def get_cluster_id(self):
+        """
+        Get cluster UUID
+        Not relying on metadata.json as user sometimes want to run
+        only with kubeconfig for some tests
+
+        Returns:
+            str: cluster UUID
+
+        """
+        cluster_id = run_cmd(
+            "oc get clusterversion version -o jsonpath='{.spec.clusterID}'"
+        )
+        return cluster_id
+
+    def get_vault_cli(self):
+        """
+        Download vault based on platform
+        basically for CLI purpose
+
+        """
+        if distro.linux_distribution == "CentOS Linux":
+            deps = " ".join(constants.VAULT_CENTOS_DEPS)
+            cmd = f"sudo yum install -y {deps} "
+            run_cmd(cmd)
+            cmd = (
+                f"sudo yum-config-manager --add-repo " f"{constants.VAULT_CENTOS_REPO}"
```

Comment:
We should not run sudo commands in ocs-ci - we should use binary if possible if not - we have to have requirements set for ocs-ci for KMS vault to have this installed before run ocs-ci.

Reply:
@petr-balogh ack, I have removed this yum based installation, instead we will be downloading the binary.